frontend/src/routes/Components/InputComponent.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from "react";
+import Style from "style-it";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+class InputComponent extends Component {
+  render() {
+    const styles = `
+    .input-group {
+      display: flex;
+     }
+     .input-group-text {
+      background-color: var(--toast-neutral-6);
+      border-radius: 20rem;
+      color: var(--toast-neutral-1);
+      padding: 0rem 0.75rem;
+    }
+     .form-control {
+       border-radius: 20rem;
+       background-color: var(--toast-neutral-6);
+       font-size: 0.96rem;
+     }
+     .required .control-label:after {
+      content: "*";
+      color: var(--toast-red);
+    }
+    `;
+
+    function hasIcon(props) {
+      if (props.icon) {
+        return (
+          <div className="input-group-prepend">
+            <span className="input-group-text" id="basic-addon"> <FontAwesomeIcon icon={props.icon} /></span>
+          </div>
+        );
+      }
+    }
+
+
+    function isRequired(props){
+      if(props.required){
+        return(
+        <div className="required">
+          {inputGroup(props)}
+          </div>
+        );
+      }
+      else{
+        return(
+        inputGroup(props)
+        );
+      }
+    }
+
+    function inputGroup(props){
```

Comment:
instead of using isRequired(), just have an array for the classes and add the classes in based on which props are there. check what i did for the ToastButton component, i think you can use the same logic here and all you need is inputGroup() and hasIcon()